### DISCUSSION TITLE
I am attempting a regression task where my labels are continuous values. However, MMPretrain always converts them to integers.

### DISCUSSION THREAD

**Discussion by rainyNighti** [open-mmlab/mmclassification#1884] [Q&A]
My labels are continuous values ranging from 1 to 10, such as 5.7. I have created a custom MyDataset(BaseDataset) that returns `{'img_path': 'path', 'gt_label': float num}`. However, after packaging the DataLoader with MMPretrain, the gt_label is converted to an integer.
I'm trying to debug the mmengin.runner.build_dataloader function, specifically until line 1476:
```
data_loader = DataLoader(
    dataset=dataset,
    sampler=sampler if batch_sampler is None else None,
    batch_sampler=batch_sampler,
    collate_fn=collate_fn,
    worker_init_fn=init_fn,
    **dataloader_cfg)
```
Before this point, the dataset contains normal decimal values for the **gt_label**. However, after packaging it with DataLoader and obtaining the gt_label through batch_sampler, the values become integers. I suspect that the collate_fn might be performing the conversion, but I'm unable to further examine the definition of default_collate. Could you please advise me on **how to keep the labels as decimal values or if MMPretrain supports decimal labels**?

I have implemented a custom MyDataset, MSE loss, MSE evaluation metric, and used the resnet18_cifar model with the num_classes in the head set to 1. Here is a partial definition:

The **MyDataset** is defined as follows：
```
@DATASETS.register_module()
class Catacrat(BaseDataset):
    def __init__(self,
                 data_root: str = '',
                 label_path: str = '',
                 split: str = 'train',
                 **kwargs):
        self.split = split
        self.label_path = label_path
        self.df = pd.read_csv(self.label_path, low_memory=False)

        super().__init__(data_root=data_root, 
                         ann_file='',
                         **kwargs)

    def load_data_list(self):

        root_path = os.path.join(self.data_root, self.split)
        img_names = os.listdir(root_path)
        img_paths = [os.path.join(root_path, x) for x in img_names]

        gt_labels = []
        for name in img_names:
            score = self.df.loc[self.df['filename'] == name, 'score'].values[0]
            gt_labels.append(float(score))

        data_list = []
        for img_path, gt_label in zip(img_paths, gt_labels):
            info = {'img_path': img_path, 'gt_label': torch.tensor(float(gt_label))}
            data_list.append(info)
        return data_list
```

**MSELoss：**
```
@MODELS.register_module()
class MSELoss(nn.Module):
    def forward(self,
                cls_score,
                label,
                avg_factor=None):
        mse_loss = F.mse_loss(cls_score.squeeze(1), label.float())
        return mse_loss
```

**Model Config:**
```
_base_ = [
    '../_base_/models/resnet18_cifar.py', '../_base_/datasets/cataract_bs16.py',
    '../_base_/schedules/cifar10_bs128.py', '../_base_/default_runtime.py'
]

model = dict(
    head=dict(
        type='LinearClsHead',
        num_classes=1,
        in_channels=512,
        loss=dict(type='MSELoss', _delete_=True),
    ))
```

**Comment by OpenByteDev** [comment]
Did you solve this?

**Comment by Patinetilla** [comment]
Any hint or tutorial on how to do it? Many thanks!
